STYLE_GUIDE.md

```diff
@@ -0,0 +1,218 @@
+
+### Variables
+
+We use Sass variables to declare colours and reuse values wherever useful. It's not necessary to declare every CSS unit value as a variable, but if a value is used in multiple places (for related reasons), then it makes sense to make it a variable - either locally in the component, or application-wide in `src/styles/_variables.scss`. Using a named variable instead of a number clarifies intent, and helps avoid updates to the value in multiple places later on.
+
+Avoid:
+
+```scss
+color: #001521;
+color: rgba(#001521, 0.5);
+color: rgba(0, 0, 0, 0.5);
+```
+
+Prefer:
+
+```scss
+color: $color-bg-dark-1;
+color: rgba($color-bg-dark-1, 0.5);
+color: rgba(black, 0.5);
+```
+
+### Theme colours
+
+Kedro-Viz has both light and dark themes, and every component must be styled appropriately for each theme. The themes are set using the same theme classes that Kedro-UI uses - `kui-theme--light` and `kui-theme--dark` - applied on the top-level container element. To simplify code and avoid increasing CSS specificity, we use CSS custom properties to handle theme colours. We recommend that you use these too where possible - they are defined in the App component.
+
+Avoid:
+
+```scss
+.pipeline-title {
+  .kui-theme--light & {
+    background: $color-bg-light-1;
+  }
+
+  .kui-theme--dark & {
+    background: $color-bg-dark-1;
+  }
+}
+```
+
+Prefer:
+
+```scss
+.kui-theme--light {
+  --color-bg-1: $color-bg-light-1;
+}
+
+.kui-theme--dark {
+  --color-bg-1: $color-bg-dark-1;
+}
+
+.pipeline-title {
+  background: var(--color-bg-1);
+}
+```
+
+## JavaScript
+
+### Imports
+
+Although it works in regular development (because this project uses React-Scripts), you should avoid importing non-standard file-types like `.scss` and `.svg` directly into JavaScript files. This is because these won't work without specific webpack loaders, and breaks when Kedro-Viz is imported into other projects. The `lib-test` testing suite exists partly for this reason, to check that Kedro-Viz still works when imported as a component library into a fairly standard JS app.
+
+### Testing
+
+In accordance with McKinsey standards, we aim to maintain at least a test coverage of at least 70% averaged across the codebase. As it's not practical to write tests for every JavaScript file, we usually aim for closer to 90% on most files, so that the project average continues to exceed the expected minimum standard.
+
+We use [Jest](https://jestjs.io/), [Enzyme](https://enzymejs.github.io/enzyme/), and [Testing Library](https://testing-library.com/) for JavaScript testing. Most of the older tests are written with Enzyme, but we are beginning to write more tests using Testing Library as it is more flexible for testing certain browser APIs. Either is acceptable. To help with mocking the Redux store, Enzyme helper utilities can be found in `/src/utils/state.mock.js`.
```

Comment:
I think some discussion of testing strategy and prioritisation would be worth adding here at some point.

Reply:
I agree, but can you expand on this? What would you say?